soc/ti/k3/am6x/r5/arm_mpu_regions.c

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2025 Texas Instruments
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/devicetree.h>
+#include <zephyr/kernel.h>
+#include <zephyr/arch/arm/mpu/arm_mpu.h>
+
+static const struct arm_mpu_region mpu_regions[] = {
+#if defined CONFIG_SOC_AM2434_R5F0_0
```

Comment:
Could you use the more generic `SOC_SERIES_AM6X_R5` here? Otherwise you will need to add this same check for all 4 R5F cores when those are added.

Reply:
Yes, this is a valid issue, right now there is no good way to distinguish between different cores since they are essentially seen as different "socs". We will need more discussion surrounding this.

Reply:
@glneo I was thinking that perhaps we could split the Kconfig for different SoCs in `soc/` directory, that would be cleaner imo and would also give us freedom to define to define an option like `CONFIG_SOC_AM2434_R5` which would be selected for all cores. Currently the closest thing to this is `CONFIG_SOC` which is a string, hence not really of use with the preprocessor.